### ISSUE TITLE
Always disable statement decomposer on C++ "auto" expressions

### ISSUE THREAD

**Issue by tiagolascasas** [specs-feup/clava#140]
If you apply the statement decomposer to C++ `auto` expressions, e.g., `auto start = high_resolution_clock::now();`, it will decompose it into `auto start; start = high_resolution_clock::now();`, which is not valid C++ code. `auto` always expects an initialization in the same statement in order to infer the type, and this is broken by detaching the declaration from the initialization.

I think it is not reasonable to expect a developer to filter out these expressions before applying the decomposer, and that instead the decomposer should be the one doing the filtering. A more advanced solution would be to keep the decomposition but replace `auto` by its intended type, although that would require significant effort.

**Comment by joaobispo** [comment]
Yes, the decomposer should have a check that skips declarations that are auto.

Also, as suggested, there could be a normalization step that transforms the auto type in the type of the initialization. Tentative code for that transformation could be:

```
for(const varDecl of Query.search("vardecl")) {
    if(varDecl.type.isAuto) {
        varDecl.type = varDecl.init.type;
    }
}
```

**Comment by lm-sousa** [comment]
Has this been fixed?
